controllers/socket.go

```diff
@@ -16,4 +22,95 @@ func SocketInit(so socketio.Socket) {
 	so.On("disconnection", func() {
 		log.Println("on disconnect")
 	})
+	so.On("createLobby", func(jsonstr string, response func(interface{}) interface{}) {
+		js, _ := simplejson.NewFromReader(strings.NewReader(jsonstr))
+
+		mapName, _ := js.Get("mapName").String()
+		format, _ := js.Get("format").String()
+		//server, _ := js.Get("server").String()
+		//rconPwd, _ := js.Get("rconpwd").String()
+		whitelist, _ := js.Get("whitelist").Int()
+
+		var playermap = map[string]lobby.LobbyType{
+			"sixes":      lobby.LobbyTypeSixes,
+			"highlander": lobby.LobbyTypeHighlander,
+		}
+
+		//TODO: Configure server here
+		lob := lobby.New(mapName, playermap[format], whitelist)
+		err := lob.Save()
+
+		if err != nil {
+			//TODO: Add stuff here
+		}
+
+		lobby_id := simplejson.New()
+		lobby_id.Set("id", string(lob.Id))
+		bytes, _ := buildSuccessJSON(lobby_id).Encode()
+		response(string(bytes))
+	})
+	so.On("addPlayer", func(jsonstr string, response func(interface{}) interface{}) {
+		js, _ := simplejson.NewFromReader(strings.NewReader(jsonstr))
+
+		//TODO: Use websockets session code for getting Player
+		//something like session.Values["steamid"]
+		var player *models.Player
+
+		slot, _ := js.Get("slot").Int()
+		lobbyidstring, _ := js.Get("lobbyid").String()
+		var lob *lobby.Lobby
+		var bytes []byte
+
+		//schalla is evil, assume he'll send us all sorts of stuff
+		//check lobbyidstring
+		if !bson.IsObjectIdHex(lobbyidstring) {
+			bytes, _ = buildFailureJSON("lobbyid is not a valid hex representation", -2).Encode()
+		} else {
+			lobbyid := bson.ObjectId(lobbyidstring)
+			query := database.GetLobbiesCollection().FindId(lobbyid)
+
+			if query == nil {
```

Comment:
here, query can't be `nil` because it still hasn't been executed. You should do `err := database.GetLobbiesCollection().FindId(lobbyid).One(&lob); if err != nil { error lobby not found}`